elements/pl-matrix-component-input/pl-matrix-component-input.mustache

```diff
@@ -0,0 +1,101 @@
+
+{{#question}}
+<script>
+    $(function() {
+        $('#pl-matrix-components-input-{{uuid}} [data-toggle="popover"]').popover({
+            container: 'body',
+            template: '<div class="popover pl-matrix-components-input-popover" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
+        });
+    });
+</script>
+<div id="pl-matrix-components-input-{{uuid}}" class="input-group">
+    <span class="input-group-prepend">
+        {{#label}}<label class="input-group-text" id="basic-addon2">{{label}}</label>{{/label}}
+    </span>
+    &nbsp;{{{input_array}}}&nbsp;
+    <div class="input-group-append">
+        <a class="btn btn-light border" role="button" data-toggle="popover" data-html="true" title="Matrix" data-content="{{info}}" data-placement="auto" data-trigger="focus" tabindex="0">
+            <i class="fa fa-question-circle" aria-hidden="true"></i>
+            {{#correct}}&nbsp;<span class="badge badge-success"><i class="fa fa-check" aria-hidden="true"></i> 100%</span>{{/correct}}
+            {{#partial}}&nbsp;<span class="badge badge-warning"><i class="fa fa-circle-o" aria-hidden="true"></i> {{partial}}%</span>{{/partial}}
+            {{#incorrect}}&nbsp;<span class="badge badge-danger"><i class="fa fa-times" aria-hidden="true"></i> 0%</span>{{/incorrect}}
+
+        </a>
+    </div>
+</div>
+{{/question}}
+
+{{#submission}}
+{{#parse_error}}
+<script>
+    $(function(){
+        $('#pl-matrix-component-input-submission-{{uuid}} [data-toggle="popover"]').popover({
+            container: 'body',
+            template: '<div class="popover pl-matrix-component-input-popover" role="tooltip"><div class="arrow"></div><h3 class="popover-header"></h3><div class="popover-body"></div></div>',
+        });
+    });
+</script>
+<span id="pl-matrix-component-input-submission-{{uuid}}">
+    {{#label}}<span>{{label}}</span>{{/label}}
+    <a class="btn btn-secondary btn-sm" role="button" data-placement="auto" data-trigger="focus" data-toggle="popover" data-html="true" title="Error" tabindex="0" data-content="{{parse_error}}{{#raw_submitted_answer}}<hr><h5><small class='text-uppercase'>What you submitted</small></h5><pre>{{raw_submitted_answer}}</pre>{{/raw_submitted_answer}}">INVALID
```

Comment:
The display of the raw submitted answer didn't work for me when I submitted invalid input.

Reply:
I found another bug when I was running a different example, that may fix the problem you found. However, I was not able to get errors when submitting invalid answers in the gradient question. Do you remember which type of invalid answer you tried?

Reply:
Here's an example below. This is tested in the instructor question view, in case that matters.

![2018-06-19 at 17 34](https://user-images.githubusercontent.com/875833/41627750-2c753336-73e7-11e8-82dd-8f969ee2347a.png)

Reply:
I had done it intentionally (can't remember the exact reason :-)) I think I was having difficulties printing the submitted matrix in latex format when one of the entries is left blank (but okay to print when it is a string like the example above). I will take a look at this again.

Reply:
For printing the raw submitted answer I think it might be better to not use latex at all, and just print out the raw data in a <pre> block or something? I'm wondering about something like:

<pre>
element 1,1: "3"
element 1,2: "5.3"
element 2,1: "" INVALID BLANK ENTRY
element 2,2: "4"
element 3,1: "sdlfkj" INVALID ENTRY
element 3,2: "5,2" INVALID ENTRY
</pre>

The main point here is not to look pretty, but to help a student realize when they've entered an invalid character (such as the comma in 5,2 in the last entry).

Anyway, this is just an idea. Anything else that clearly shows all the entered data and which one is wrong would also be fine.

I think printing out exactly what the student entered is helpful for both the student and a TA or instructor looking at the problem later (although they can also view the raw submission data in the event log).

Reply:
thanks for the input. I will work on it!

Reply:
I added the detailed description of all entries as suggested above when there is an invalid answer. If all answers are valid, I decided to keep the display using latex format (I think it is cleaner). Also, I am only giving feedback if each entry is correct or not when `allow-feedback=true`.

Reply:
Two more quick things:

1) I suggest putting the invalid raw submitted answer inside the "invalid" popover - this is what's done in both `pl-number-input` and `pl-matrix-input`, for example.

2) I suggest displaying the invalid raw submitted answer (inside the popover) in table format, with the exact same code that's used to create the array of inputs in the question panel - then you wouldn't have to write "element 3, 1" or whatever, wouldn't have to put the submission in quotes, wouldn't have to rely on students knowing that "3, 1" meant "third column and first row," etc. It seems like this would be easier to interpret?